### DISCUSSION TITLE
icon association for mime types with prefixes in glob pattern field

### DISCUSSION THREAD

**Discussion by ciros88** [doublecmd/doublecmd#1014] [General]
double commander seems the only file manager i know that displays the direct mime type associated icon instead of the audio/x-generic one. the problem is that the icon is not displayed for those mime types that have a prefix wilcard in the glob pattern field.

an example:
![image](https://github.com/doublecmd/doublecmd/assets/84340161/1f919e1f-a9ff-4a1f-84b6-b8e51681fe3f)
here, the mime declares both prefixes and suffixes pattern: the mimetypes association works, i mean if i click on a file that matches any on these patterns the associated app will be successfully launched (both prefixes and suffixes patterns).
i have also associated a custom icon (the green one with the white B inside) to the mime type, but seems to works only for suffixes patterns.
this is an example:
![image](https://github.com/doublecmd/doublecmd/assets/84340161/a987adbe-a2ce-4420-8626-2b5928ff3cc3)
any hint? thanks

**Comment by alexx2000** [comment]
DC works only with file extensions (suffixes). DC creates mapping: file extension -> icon name.

**Comment by ciros88** [comment]
Thanks for the quick answer. Why not support prefixes also? Since it is featured by xdg on linux

**Reply by ciros88** [reply]
i've noticed that reproducing the above mentioned scenario with others filemanagers in linux (tried dolphin, krusader, nautilus, thunar, pcmanfm and a couple of others) the xdg mime type prefix association is supported (showing the right audio-x-generic icon in this specific case).

**Comment by alexx2000** [comment]
Because DC works with file extensions only.

**Reply by ciros88** [reply]
do you think there is a chance to embrace the xdg mime standard instead?

**Comment by ciros88** [comment]
double commander seems also to ignore the icon association overrides for specific users, but handle only the system ones.

doing the following:

`sudo xdg-desktop-menu install --novendor --mode system "$path_to_myapp_desktop_file"`

..then creating a custom mime type xml file in the /usr/share/mime/packages specifying the icon as audio-x-generic (sorry for the wrong indentation):
`<?xml version="1.0" encoding="utf-8"?>
<mime-info xmlns="http://www.freedesktop.org/standards/shared-mime-info">
  <mime-type type="audio/x-ahx">
    <comment>Abyss' Highest eXperience</comment>
    <generic-icon name="audio-x-generic"/>
    <icon name="audio-x-generic"/>
      <magic>
        <match type="big32" value="0x54485800" offset="0"/>
        <match type="big32" value="0x54485801" offset="0"/>
      </magic>
    <glob-deleteall/>
    <glob pattern="ahx.*"/>
    <glob pattern="*.ahx"/>
  </mime-type>
</mime-info>`

`sudo xdg-mime default "$myapp_desktop_filename" "$just_created_mime_type_to_associate_with"`
`sudo update-mime-database /usr/share/mime`
`sudo update-desktop-database /usr/share/applications`

will result in a working association for the specified mime with myapp (it will be launched clicking on a matching file), displaying the right audio-x-generic icon in every file manager including double commander.

instead, doing the same but for a specific user (user xdg mimes normally overrides system ones) as the following:

`xdg-desktop-menu install --novendor --mode user "$path_to_myapp_desktop_file"`

..then creating the same custom mime type xml file, but in the $HOME/.local/share/mime/packages specifying the icon as audio-x-generic:
`<?xml version="1.0" encoding="utf-8"?>
<mime-info xmlns="http://www.freedesktop.org/standards/shared-mime-info">
  <mime-type type="audio/x-ahx">
    <comment>Abyss' Highest eXperience</comment>
    <generic-icon name="audio-x-generic"/>
    <icon name="audio-x-generic"/>
      <magic>
        <match type="big32" value="0x54485800" offset="0"/>
        <match type="big32" value="0x54485801" offset="0"/>
      </magic>
    <glob-deleteall/>
    <glob pattern="ahx.*"/>
    <glob pattern="*.ahx"/>
  </mime-type>
</mime-info>`

`xdg-mime default "$myapp_desktop_filename" "$just_created_mime_type_to_associate_with"`
`update-mime-database "$HOME"/.local/share/mime`
`update-desktop-database "$HOME"/.local/share/applications`

will result in a working association for the specified mime with myapp (for every file manager), but the right audio-x-generic icon is correctly displayed in every file manager Excluding double commander. double cmd will display, instead, the system mime icon association or, if it doesn't exists, will display the question mark "?" icon, completely ignoring the user mime icon association.
I think this is very limiting, and opening a bug/enhancement ticket is somewhat reasonable since user level mime icon association is a very common scenario

**Reply by ciros88** [reply]
@alexx2000 what do you think about?

**Reply by alexx2000** [reply]
See #637, it is a same issue.

**Reply by ciros88** [reply]
#637 issue is related to https://github.com/doublecmd/doublecmd/discussions/1014#discussion-5282517 but not https://github.com/doublecmd/doublecmd/discussions/1014#discussioncomment-6248904 . Here i'm pointing to a full xdg mime support, something more comprehensive than #637

**Reply by alexx2000** [reply]
#637 same as https://github.com/doublecmd/doublecmd/discussions/1014#discussioncomment-6248904
